rfd/0059-search-based-access-requests.md

```diff
@@ -0,0 +1,276 @@
+---
+authors: Alexander Klizhentas (sasha@goteleport.com), Nic Klaassen (nic@goteleport.com)
+state: draft
+---
+
+# RFD 59 - Search Based Access Requests
+
+## What
+
+This RFD proposes just-in-time search-based access requests which will enable
+users to request access to resources based on a search and/or selection of
+individual resources, rather than having to request one or more roles.
+
+## Why
+
+Teleport users may not know in advance which roles they need or who should grant
+access, just which resource(s) they need access to.
+
+Users would like to avoid having many different custom roles for different
+levels of access.
+
+Requesting access to a "blanket" role violates the principle of least priviledge.
+Most of the time incident responders only need access to 1 or 2 nodes,
+search-based access requests will allow them to request access to only the nodes
+they need.
```

Comment:
I think this is well stated: there is a further point in here that a 'review' of a blanket access request is enormously harder than a specific request. 

if Alice has the role of splunk-admin, she may have access to tens of thousands of systems. Its extremely easy to generate a reason to login to one of them, but then connect to a different one, or all of them. Permitting access to a single box to troubleshoot (say) Splunk is very, very different to permitting somebody to run Ansible across the fleet. If we step back and consider that the concern is not just Alice suddenly turning malicious, but that an attacker could get her (short lived) secret, it makes no sense for Alice to have a secret that allows her to connect to all of prod to fix Splunk on a single one.